WiserTaskScheduler/WiserTaskScheduler/Modules/ServerMonitors/Services/ServerMonitorsService.cs

```diff
@@ -0,0 +1,443 @@
+
+            //Set the email settings correctly.
+            receiver = monitorItem.EmailAddressForWarning;
+            subject = "CPU usage too high.";
+            body = $"The CPU usage has been above the threshold of {threshold}% for {aboveThresholdTimer} runs.";
+            await logService.LogInformation(logger, LogScopes.RunStartAndStop, monitorItem.LogSettings, $"CPU is: {realvalue}%", configurationServiceName, monitorItem.TimeId, monitorItem.Order);
+
+            //Checks if the value is above the threshold and if so then adds 1 to the counter.
+            //If the value is below the threshold then reset the counter to 0.
+            if(realvalue < threshold)
+            {
+                emailCpuSent = false;
+                aboveThresholdTimer = 0;
+                return;
+            }
+
+            //Check if an email already has been sent
+            if (emailCpuSent)
+            {
+                return;
+            }
+
+            aboveThresholdTimer++;
+            if (aboveThresholdTimer >= 6)
+            {
+                emailCpuSent = true;
+                await gclCommunicationsService.SendEmailAsync(receiver, subject, body);
+            }
+
+            await logService.LogInformation(logger, LogScopes.RunStartAndStop, monitorItem.LogSettings, $"CPU timer count  is: {aboveThresholdTimer}", configurationServiceName, monitorItem.TimeId, monitorItem.Order);
+        }
+
+        /// <summary>
+        /// Gets the network utilization of the specified network card and checks if an e-mail needs to be send.
+        /// </summary>
+        /// <param name="monitorItem">The information for the monitor action.</param>
+        /// <param name="threshold">The threshold to check if an email needs to be send.</param>
+        /// <param name="gclCommunicationsService">The name of the service in the configuration, used for logging.</param>
+        /// <param name="configurationServiceName">The communications service from the GCL to actually send out the email.</param>
+        /// <returns></returns>
+        private async Task GetNetworkUtilization(ServerMonitorModel monitorItem, int threshold, string configurationServiceName, CommunicationsService gclCommunicationsService)
+        {
+            var networkInterfaceName = monitorItem.NetworkInterfaceName;
+
+            const int numberOfIterations = 10;
+
+            //get the correct types of the performancecounter class
+            PerformanceCounter bandwidthCounter = new PerformanceCounter("Network Interface", "Current Bandwidth", networkInterfaceName);
+            var bandwidth = bandwidthCounter.NextValue();
+            PerformanceCounter dataSentCounter = new PerformanceCounter("Network Interface", "Bytes Sent/sec", networkInterfaceName);
+            PerformanceCounter dataReceivedCounter = new PerformanceCounter("Network Interface", "Bytes Received/sec", networkInterfaceName);
+
+            float sendSum = 0;
```

Comment:
De floats mogen als var gezet worden, door 0f te doen geef je aan dat het een floating number is.